meerk40t/main.py

```diff
@@ -123,179 +123,182 @@ def run():
         pass
+    else:
+        kernel.add_plugin(inputs.plugin)
 
     try:
         from .core import elements
-
-        kernel.add_plugin(elements.plugin)
-    except ImportError:
+    except Mk40tImportAbort:
         pass
+    else:
+        kernel.add_plugin(elements.plugin)
 
     try:
         from .core import bindalias
-
-        kernel.add_plugin(bindalias.plugin)
-    except ImportError:
+    except Mk40tImportAbort:
         pass
+    else:
+        kernel.add_plugin(bindalias.plugin)
 
     try:
         from .core import webhelp
-
-        kernel.add_plugin(webhelp.plugin)
-    except ImportError:
+    except Mk40tImportAbort:
         pass
+    else:
+        kernel.add_plugin(webhelp.plugin)
 
     try:
         from .core import planner
-
-        kernel.add_plugin(planner.plugin)
-    except ImportError:
+    except Mk40tImportAbort:
         pass
+    else:
+        kernel.add_plugin(planner.plugin)
 
     try:
         from .image import imagetools
-
-        kernel.add_plugin(imagetools.plugin)
-    except ImportError:
+    except Mk40tImportAbort:
         pass
+    else:
+        kernel.add_plugin(imagetools.plugin)
 
     try:
         from .device.lhystudios import lhystudiosdevice
-
-        kernel.add_plugin(lhystudiosdevice.plugin)
-    except ImportError:
+    except Mk40tImportAbort:
         pass
+    else:
+        kernel.add_plugin(lhystudiosdevice.plugin)
 
     try:
         from .device.moshi import moshidevice
-
-        kernel.add_plugin(moshidevice.plugin)
-    except ImportError:
+    except Mk40tImportAbort:
         pass
+    else:
+        kernel.add_plugin(moshidevice.plugin)
 
     try:
         from .device.grbl import grbldevice
-
-        kernel.add_plugin(grbldevice.plugin)
-    except ImportError:
+    except Mk40tImportAbort:
         pass
+    else:
+        kernel.add_plugin(grbldevice.plugin)
 
     try:
         from .device.ruida import ruidadevice
-
-        kernel.add_plugin(ruidadevice.plugin)
-    except ImportError:
+    except Mk40tImportAbort:
         pass
+    else:
+        kernel.add_plugin(ruidadevice.plugin)
 
     try:
         from .core import svg_io
-
-        kernel.add_plugin(svg_io.plugin)
-    except ImportError:
+    except Mk40tImportAbort:
         pass
+    else:
+        kernel.add_plugin(svg_io.plugin)
 
     try:
         from .extra import vectrace
-
-        kernel.add_plugin(vectrace.plugin)
-    except ImportError:
+    except Mk40tImportAbort:
         pass
+    else:
+        kernel.add_plugin(vectrace.plugin)
 
     try:
         from .extra import inkscape
-
-        kernel.add_plugin(inkscape.plugin)
-    except ImportError:
+    except Mk40tImportAbort:
         pass
+    else:
+        kernel.add_plugin(inkscape.plugin)
 
     try:
         from .extra import embroider
-
-        kernel.add_plugin(embroider.plugin)
-    except ImportError:
+    except Mk40tImportAbort:
         pass
+    else:
+        kernel.add_plugin(embroider.plugin)
 
```

Comment:
With embroider I'm actually a little more inclined towards the try since I think there's little problem simply deleting that plugin file and it would just be gone. Though the hard-import for frozen would stay and fail regularly. Most of the stuff in extra is supposed somewhat failable. Since really they aren't supposed to do anything mission critical.